lib/contexts/DiceContext/DiceContext.tsx

```diff
@@ -104,22 +84,10 @@ export function useDice() {
     return roll(commandOptions, optionsForRoll);
   }
 
-  /**
-   * Reroll the latest commands
-   * @param options if empty, uses the latest options
-   */
-  function reroll(optionsForRoll: IRollDiceOptions = options) {
-    const result = Dice.rollCommandOptionList(
-      latestCommandOptions,
-      optionsForRoll
-    );
-    return result;
-  }
-
   function addOrRemovePoolElement(element: IDicePoolElement) {
     const isFirstPoolElement = pool.length === 0;
     if (isFirstPoolElement) {
-      setCommandGroups([]);
+      preparePool();
     }
```

Comment:
instead of clearing the command groups, we save the currently selected commands to re-select them once the pool is rolled